webapp/channels/src/components/admin_console/system_users/system_users.tsx

```diff
@@ -190,6 +200,56 @@ export class SystemUsers extends React.PureComponent<Props, State> {
         this.setState({loading: false});
     };
 
+    onSearch = async (term: string) => {
+        this.setState({loading: true});
+
+        const options = {
+            ...searchUserOptionsFromFilter(this.props.filter),
+            ...this.props.teamId && {team_id: this.props.teamId},
+            ...this.props.teamId === SearchUserTeamFilter.NO_TEAM && {
+                [UserSearchOptions.WITHOUT_TEAM]: true,
+            },
+            allow_inactive: true,
+        };
+
+        const {data: profiles} = await this.props.actions.searchProfiles(term, options);
+        if (profiles.length === 0 && term.length === USER_ID_LENGTH) {
+            await this.getUserByTokenOrId(term);
+        }
+
+        this.setState({loading: false});
+    };
+
+    onFilter = async ({teamId, filter}: {teamId?: string; filter?: string}) => {
+        if (this.props.searchTerm) {
+            this.onSearch(this.props.searchTerm);
+            return;
+        }
+
+        this.setState({loading: true});
+
+        const newTeamId = typeof teamId === 'undefined' ? this.props.teamId : teamId;
```

Comment:
Can we do a coalesce here, or can the return value be `null`?
```suggestion
        const newTeamId = teamId ?? this.props.teamId;
```

Reply:
No we cant because the filter can come as empty string, so only way here is to check for undefined